Sources/DarwinPlatform/Darwin.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import SwiftlyCore
+@_implementationOnly import Archive
+
+/// `Platform` implementation for Darwin systems.
+/// This implementation can be reused for any supported Darwin platform.
+/// TODO: replace dummy implementations
+public struct Darwin: Platform {
+    public init() {}
+
+    public var appDataDirectory: URL {
+        if let dir = ProcessInfo.processInfo.environment["XDG_DATA_HOME"] {
+            return URL(fileURLWithPath: dir)
+        } else {
+            return FileManager.default.homeDirectoryForCurrentUser
+                .appendingPathComponent(".local", isDirectory: true)
+                .appendingPathComponent("share", isDirectory: true)
+        }
+    }
+
+    public var toolchainFileExtension: String {
+        "pkg"
+    }
+
+    public func isSystemDependencyPresent(_: SystemDependency) -> Bool {
+        true
+    }
+
+    public func install(from tmpFile: URL, version: ToolchainVersion) throws {
+        guard tmpFile.fileExists() else {
+            throw Error(message: "\(tmpFile) doesn't exist")
+        }
+
+        if !self.swiftlyToolchainsDir.fileExists() {
+            try FileManager.default.createDirectory(at: self.swiftlyToolchainsDir, withIntermediateDirectories: false)
+        }
+
+        SwiftlyCore.print("Extracting toolchain...")
+        let toolchainDir = self.swiftlyToolchainsDir.appendingPathComponent(version.name)
```

Comment:
Should toolchain directory be `~/Library/Developer/Toolchains` as this is the standard local directory to use for Toolchain installs with Xcode

Reply:
Yes. We should have done it. The implementation of `DarwinPlatform` is mostly duplicated from `LinuxPlatform`. I'd like to leave it to a future PR/improvement.(This PR only promise we can build it on macOS platform instead of making it usable on macOS)
But if you think the current code status does not meet the merge standard, I can also address it in this PR.